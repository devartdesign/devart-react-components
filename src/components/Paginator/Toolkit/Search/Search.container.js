import { connect } from 'react-redux';

import { paginatorState } from '../../../../../state-mgmt';
import Search from './Search';

export const mapStateToProps = state => ({
  search: state.paginator.search
});

export const mapDispatchToProps = dispatch => ({
  onSearch: text => dispatch(paginatorState.actions.onSearch(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Search);
