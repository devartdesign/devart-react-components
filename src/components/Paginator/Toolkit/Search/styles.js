export const styles = theme => ({
  search: {
    position: 'relative',
    borderRadius: theme.shape.borderRadius,
    backgroundColor: theme.palette.common.white,
    '&:hover': {
      backgroundColor: theme.palette.common.white
    },
    marginLeft: 0,
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      marginLeft: theme.spacing.unit,
      width: 'auto'
    }
  },
  searchIcon: {
    width: theme.spacing.unit * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  inputRoot: {
    color: 'inherit',
    width: '100%',
  },
  inputInput: {
    paddingTop: theme.spacing.unit,
    paddingRight: theme.spacing.unit,
    paddingBottom: theme.spacing.unit,
    paddingLeft: theme.spacing.unit * 10,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('sm')]: {
      width: 120,
      '&:focus': {
        width: 200,
      },
    },
  }
});
