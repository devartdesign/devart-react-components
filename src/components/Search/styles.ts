import { StyleSheet } from 'aphrodite/no-important';
import { Color } from 'devart-utils/dist/models/general';

export default StyleSheet.create({
  search: {
    position: 'relative',
    borderRadius: 5,
    backgroundColor: Color.WHITE,
    '&:hover': {
      backgroundColor: Color.LIGHT_GRAY
    },
    marginLeft: 0,
    width: '100%',
  },
  searchIcon: {
    width: 3 * 9,
    height: '100%',
    position: 'absolute',
    pointerEvents: 'none',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center'
  },
  icon: {
    background: '../../assets/icons/search.svg'
  },
  input: {
    color: 'inherit',
    paddingTop: 3,
    paddingRight: 3,
    paddingBottom: 3,
    paddingLeft: 3 * 10,
    width: '100%',
    // transition: theme.transitions.create('width'),
  }
});
