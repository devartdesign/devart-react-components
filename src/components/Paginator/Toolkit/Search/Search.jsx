import React from 'react';
import PropTypes from 'prop-types';

import { withStyles } from '@material-ui/core/styles';
import SearchIcon from '@material-ui/icons/Search';
import Input from '@material-ui/core/Input';
import { styles } from './styles';

class Search extends React.PureComponent {
  static propTypes = {
    classes: PropTypes.object,
    onSearch: PropTypes.func.isRequired,
    search: PropTypes.string
  };

  static defaultProps = {
    classes: {},
    search: ''
  }

  handleChange = (event) => {
    event.preventDefault();
    this.props.onSearch(event.target.value);
  };

  render() {
    const { classes, search } = this.props;

    return (
      <div className={classes.search}>
        <div className={classes.searchIcon}>
          <SearchIcon />
        </div>
        <Input
          name="search"
          placeholder="Searchâ€¦"
          value={search}
          onChange={this.handleChange}
          classes={{
            root: classes.inputRoot,
            input: classes.inputInput,
          }}
        />
      </div>
    );
  }
}

export default withStyles(styles)(Search);
