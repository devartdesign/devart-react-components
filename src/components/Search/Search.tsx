import React, { memo, useCallback, useState, useEffect } from 'react';
import { css } from 'aphrodite/no-important';

import { SearchIcon } from '../../icons';
import styles from './styles';

export interface ISearchProps {
  onSearch: (searchText: string) => void;
}

const Search = ({ onSearch }: ISearchProps) => {
  const [state, setState] = useState('');
  const handleChange = useCallback((event) => {
    event.preventDefault();
    setState(event.target.value);
  }, []);

  useEffect(() => { onSearch(state); }, [onSearch, state]);
  return (
    <div className={css(styles.search)}>
      <div className={css(styles.searchIcon)}>
        <SearchIcon />
      </div>
      <input
        type="text"
        name="search"
        placeholder="Searchâ€¦"
        value={state}
        onChange={handleChange}
        className={css(styles.input)}
      />
    </div>
  );
};

export default memo(Search);
