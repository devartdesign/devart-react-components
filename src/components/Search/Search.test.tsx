import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Search, { ISearchProps } from './Search';

describe('<Search />', () => {
  let wrapper;
  const defaultProps: ISearchProps = {
    onSearch: jest.fn()
  };

  describe('render', () => {
    beforeEach(() => {
      wrapper = shallow(<Search {...defaultProps} />);
    });

    it('should render <SearchIcon> and an <input> components', () => {
      expect(renderer.create(wrapper)
        .toJSON())
        .toMatchSnapshot();
    });
  });

  describe('handleChange', () => {
    beforeEach(() => {
      wrapper = shallow(<Search {...defaultProps} />);
      wrapper.dive().instance().handleChange({
        preventDefault: () => { /** */},
        target: {
          name: 'search',
          value: 'search me'
        }
      });
    });

    it('should call onSearch callback from Props', () => {
      expect(defaultProps.onSearch).toHaveBeenCalledWith('search me');
    });
  });
});
