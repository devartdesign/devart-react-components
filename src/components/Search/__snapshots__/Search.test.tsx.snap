// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> render should render <SearchIcon> and an <input> components 1`] = `
.search {
  position: relative;
  width: 100%;
}

.searchIcon {
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  width: 27px;
  height: 100%;
  position: absolute;
  pointer-events: none;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  -webkit-align-items: center;
  align-items: center;
  -webkit-justify-content: center;
  justify-content: center;
  margin-left: 5px;
  fill: #29434e;
}

.input {
  color: inherit;
  padding-top: 3px;
  padding-right: 3px;
  padding-bottom: 3px;
  padding-left: 33px;
  width: 100%;
  border-radius: 3px;
  border: 1px solid #eee;
  font-size: 14px;
}

<div
  className="search"
>
  <div
    className="searchIcon"
  >
    <svg
      height={24}
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
    </svg>
  </div>
  <input
    className="input"
    name="search"
    onChange={[Function]}
    placeholder="Searchâ€¦"
    type="text"
    value=""
  />
</div>
`;
